libs/common/src/utils/nonce.service.ts

```diff
@@ -0,0 +1,57 @@
+import { InjectRedis } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger, OnApplicationBootstrap } from '@nestjs/common';
+import Redis from 'ioredis';
+import fs from 'fs';
+import { createKeys } from '../blockchain/create-keys';
+import { RedisUtils } from './redis';
+import { BlockchainService } from '../blockchain/blockchain.service';
+import { ConfigService } from '../config/config.service';
+
+@Injectable()
+export class NonceService implements OnApplicationBootstrap {
+  private logger: Logger;
+
+  private accountId: Uint8Array;
+
+  constructor(
+    @InjectRedis() private redis: Redis,
+    private blockchainService: BlockchainService,
+    private configService: ConfigService,
+  ) {
+    this.logger = new Logger(NonceService.name);
+    redis.defineCommand('incrementNonce', {
+      numberOfKeys: RedisUtils.NUMBER_OF_NONCE_KEYS_TO_CHECK,
+      lua: fs.readFileSync('lua/incrementNonce.lua', 'utf8'),
+    });
+  }
+
+  async onApplicationBootstrap() {
+    this.accountId = createKeys(this.configService.getProviderAccountSeedPhrase()).publicKey;
+    const nextNonce = await this.getNextNonce();
+    this.logger.log(`nonce is set to ${nextNonce}`);
+  }
+
+  async getNextNonce(): Promise<number> {
```

Comment:
This is the main function that returns a new nonce in every call, this also takes care of race condition when many services try to get the nonce